### ISSUE TITLE
Huge incompatibility with Divi

### ISSUE THREAD

**Issue by dchenk** [udx/wp-stateless#430]
This plugin's 'sanitize_file_name' filter totally screws up the file-based caching that Divi and the plugins Bloom and Monarch do. When the Divi theme, for example, serializes the minified CSS for a page or for the site's global styling, or when it serializes the content of a page into a binary ".data" file for caching, it constructs a potential file name for this content using a particular structure for this file name. It then does a `glob` of the potential target directory to see if there already is an existing file that can be used (previously cached). However, when Divi constructs this potential file name, it passes it through the `sanitize_file_name` function (see the constructor of the `ET_Core_PageResource` class in `Divi/core/components/PageResource.php`), which applies the 'sanitize_file_name' filter.

This means that wp-stateless prepsends this _potential_ file name with a random string, so in the call to `glob` no matching files are found. This means that, _on every single request for that page_, Divi will write the serialized data to disk after globbing for an existing cached file!

This behavior quickly used up all 20 GB of the disk where these files were being written, and it noticeably hurt performance of my websites.

I'm running the version 2.3.2 of this plugin in stateless mode. Please address this as soon as possible.

**Comment by harrystymiest** [comment]
I'm having this issue as well and it happens even with Divi's Static CSS turned off

**Comment by donbowman** [comment]
are there any workaround for this?

**Comment by harrystymiest** [comment]
> are there any workaround for this?

Until they get it going I would write a cron like this
rm -rf public_html/wp-content/et-cache and run it every 10 min

**Comment by donbowman** [comment]
its not sufficient.
it has a second side affect, some assets/css don't load properly.
so the site doesn't render after Divi 4.0.8 reliably.

**Comment by antonkorotkov** [comment]
Hello everyone. The issue has been reviewed and confirmed. We are working on it. Thanks for your patience.

**Comment by antonkorotkov** [comment]
@dchenk @harrystymiest @donbowman can you guys help me testing it? Here is the zip of the fixed version https://github.com/wpCloud/wp-stateless/archive/develop-anton.zip 
Please let me know if it works correctly (do not forget to enable the Divi support in settings) and we will include the fix into the nearest release.

Thanks.

**Comment by donbowman** [comment]
will do

**Comment by harrystymiest** [comment]
I installed on my Goddaughter's model portfolio site (Not a "client" site, Have a backup) ill monitor the size over the next 24H or so

**Comment by donbowman** [comment]
```
[Thu Feb 20 15:49:58.435424 2020] [php7:error] [pid 22437] [client 10.16.1.100:51152] PHP Fatal error:  Uncaught Error: Class name must be a valid object or a string in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php:5645\nStack trace:\n#0 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php(3231): ET_Builder_Element->_add_position_fields()\n#1 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php(425): ET_Builder_Element->_add_additional_fields()\n#2 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/main-structure-elements.php(1214): ET_Builder_Element->__construct()\n#3 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/framework.php(633): require_once('/bitnami/wordpr...')\n#4 /opt/bitnami/wordpress/wp-includes/class-wp-hook.php(288): et_builder_add_main_elements(Object(WP))\n#5 /opt/bitnami/wordpress/wp-includes/class-wp-hook.php(312): WP_Hook->apply_filters(NULL, Array)\n#6 /opt/bitnami/wordpress/wp-includes/plugin.php(544): WP_Hook->do_action(Array)\n#7 /opt/bitnami/wordpress/wp-includes/cl in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php on line 5645
[Thu Feb 20 15:50:04.652671 2020] [php7:error] [pid 22461] [client 10.16.1.100:51524] PHP Fatal error:  Cannot redeclare et_pb_hover_options() (previously declared in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/autoload.php:119) in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/module/helpers/HoverOptions.php on line 183
[Thu Feb 20 15:50:05.197791 2020] [php7:error] [pid 22416] [client 10.16.1.100:51532] PHP Fatal error:  Cannot redeclare et_pb_hover_options() (previously declared in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/autoload.php:119) in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/module/helpers/HoverOptions.php on line 183
^C
```

**Comment by harrystymiest** [comment]
No issues for me so far 
• I installed the Dev Plugin 
• I removed my cron Work around 
• Reenabled static CSS generation

**Comment by antonkorotkov** [comment]
> ```
> [Thu Feb 20 15:49:58.435424 2020] [php7:error] [pid 22437] [client 10.16.1.100:51152] PHP Fatal error:  Uncaught Error: Class name must be a valid object or a string in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php:5645\nStack trace:\n#0 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php(3231): ET_Builder_Element->_add_position_fields()\n#1 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php(425): ET_Builder_Element->_add_additional_fields()\n#2 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/main-structure-elements.php(1214): ET_Builder_Element->__construct()\n#3 /bitnami/wordpress/wp-content/themes/Divi/includes/builder/framework.php(633): require_once('/bitnami/wordpr...')\n#4 /opt/bitnami/wordpress/wp-includes/class-wp-hook.php(288): et_builder_add_main_elements(Object(WP))\n#5 /opt/bitnami/wordpress/wp-includes/class-wp-hook.php(312): WP_Hook->apply_filters(NULL, Array)\n#6 /opt/bitnami/wordpress/wp-includes/plugin.php(544): WP_Hook->do_action(Array)\n#7 /opt/bitnami/wordpress/wp-includes/cl in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/class-et-builder-element.php on line 5645
> [Thu Feb 20 15:50:04.652671 2020] [php7:error] [pid 22461] [client 10.16.1.100:51524] PHP Fatal error:  Cannot redeclare et_pb_hover_options() (previously declared in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/autoload.php:119) in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/module/helpers/HoverOptions.php on line 183
> [Thu Feb 20 15:50:05.197791 2020] [php7:error] [pid 22416] [client 10.16.1.100:51532] PHP Fatal error:  Cannot redeclare et_pb_hover_options() (previously declared in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/autoload.php:119) in /bitnami/wordpress/wp-content/themes/Divi/includes/builder/module/helpers/HoverOptions.php on line 183
> ^C
> ```

Sorry, it does not look like WP-Stateless errors. Can you try maybe re-installing the Divi?

**Comment by donbowman** [comment]
this was after the re-isntall of divi, let me try again.

i had 4.0.8 (which is last version that my site would render w/ w/ wp-stateless).
i added the dev build of wp-stateless. observed site still rendered.
i then clicked update divi
the site stopped rendering, and this error occurred.

let me try again

**Comment by donbowman** [comment]
ok the issue was the divi security patcher which was the workaround for a security flaw on 4.0.8.
remove that, update to 4.3.2, now it seems to render and not give the errors.
and the cache dir is not growing as fast.

**Comment by harrystymiest** [comment]
I'm ready to say this is resolved

**Comment by antonkorotkov** [comment]
Closing the issue. Fix will be added to the next release. Thanks.
